internal/spinlock/rwmutext.go

```diff
@@ -0,0 +1,121 @@
+package spinlock
```

Comment:
what's the difference with [sync.RWMutex](https://pkg.go.dev/sync#RWMutex) by the way?

Reply:
the original uses os semaphore